src/middlewared/middlewared/plugins/crypto.py

```diff
@@ -562,6 +566,7 @@ def create_certificate_signing_request(self, cert_info):
             Str('common'),
             Str('country'),
             Str('CSR'),
+            Str('directory_uri'),
```

Comment:
What is directory_uri? Lets Encrypt server?

Reply:
Yes - Any ACME based Cerver

Reply:
The same seems counter intuitive, can we go with something more direct? e.g. acme_server_uri ?
We should probably also provide some defaults in a method so the UI can use.

Reply:
I'll update the key's name and secondly yes I do intend to place let's encrypt staging env and production env options there and the user can type one if he/she has another one